routers/private/hook_post_receive.go

```diff
@@ -87,6 +93,46 @@ func HookPostReceive(ctx *gitea_context.PrivateContext) {
 			})
 			return
 		}
+
+		for _, update := range updates {
+			if !update.RefFullName.IsBranch() {
+				continue
+			}
+			if repo == nil {
+				repo = loadRepository(ctx, ownerName, repoName)
+				if ctx.Written() {
+					return
+				}
+				wasEmpty = repo.IsEmpty
+			}
+
+			if !update.IsDelRef() {
+				if gitRepo == nil {
+					var err error
+					gitRepo, err = gitrepo.OpenRepository(ctx, repo)
+					if err != nil {
+						log.Error("Failed to open repository: %s/%s Error: %v", ownerName, repoName, err)
+						ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
+							Err: fmt.Sprintf("Failed to open repository: %s/%s Error: %v", ownerName, repoName, err),
+						})
+						return
+					}
+				}
+				commit, err := gitRepo.GetCommit(update.NewCommitID)
+				if err != nil {
+					ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
+						Err: fmt.Sprintf("Failed to get commit %s in repository: %s/%s Error: %v", update.NewCommitID, ownerName, repoName, err),
+					})
+					return
+				}
+				if err := repo_service.SyncBranchToDB(ctx, repo.ID, update.PusherID, update.RefFullName.BranchName(), commit); err != nil {
+					ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
+						Err: fmt.Sprintf("Failed to sync branch to DB in repository: %s/%s Error: %v", ownerName, repoName, err),
+					})
+					return
+				}
+			}
```

Comment:
Can you add deletebranch here with else?

Reply:
75950d0743ef08344922c4a55be7ecab5c475e77